### ISSUE TITLE
bug in best metric selection in save checkpoint if num validation sets != 1

### ISSUE THREAD

**Issue by jbross-ibm-research** [NVIDIA-NeMo/Automodel#970]
**Describe the bug**

misplaced closing bracket `)` leads to exception if `len(val_loss) != 1`  in https://github.com/NVIDIA-NeMo/Automodel/blob/main/nemo_automodel/recipes/base_recipe.py#L211

`next()` is called on best_metric_key if len(val_loss) != 1.

```
val_loss[next(iter(val_loss.keys()) if len(val_loss) == 1 else best_metric_key)] if val_loss else None
```

Fixed version :

```
val_loss[next(iter(val_loss.keys())) if len(val_loss) == 1 else best_metric_key] if val_loss else None
```

**Steps/Code to reproduce bug**

run training with multiple validation sets and enable checkpointing

**Comment by qiaochuz-nv** [comment]
## QA RCCA Analysis

**Fix PR:** #973
**Root Cause:** Misplaced closing bracket in `base_recipe.py` best_metric_key logic — `next()` called on wrong type when `len(val_loss) != 1`.

**Coverage:** COVERED — Same code path as #971 (best_metric_key config field), already analyzed. Fix #973 is a one-line bracket correction. Upstream unit tests cover the `best_metric_key` logic.

**Conclusion:** No new test needed.
